docs/x509.rst

```diff
@@ -181,6 +181,76 @@ X.509 Certificate Object
 
         For version 3 X.509 certificates.
 
+.. class:: NameAttribute
+
+    .. versionadded:: 0.8
+
+    An X.509 name consists of a list of NameAttribute objects.
+
+    .. attribute:: oid
+
+        :type: :class:`ObjectIdentifier`
+
+        The attribute OID.
+
+    .. attribute:: value
+
+        :type: :term:`text`
+
+        The value of the attribute.
+
+.. class:: ObjectIdentifier
+
+    .. versionadded:: 0.8
+
+    Object identifiers (frequently seen abbreviated as OID) identify the type
+    of a value (see: :class:`NameAttribute`).
+
+    .. attribute:: value
+
+        :type: :class:`str`
```

Comment:
Ok, sounds good.

A different question: is value really a good name for this attribute?

On Wed, Feb 4, 2015 at 11:01 AM, Paul Kehrer <notifications@github.com>
wrote:

> In docs/x509.rst
> <https://github.com/pyca/cryptography/pull/1645#discussion_r24108981>:
>
> > +    .. attribute:: value
> > +
> > +        :type: :term:`text`
> > +
> > +        The value of the attribute.
> > +
> > +.. class:: ObjectIdentifier
> > +
> > +    .. versionadded:: 0.8
> > +
> > +    Object identifiers (frequently seen abbreviated as OID) identify the type
> > +    of a value (see: :class:`NameAttribute`).
> > +
> > +    .. attribute:: value
> > +
> > +        :type: :class:`str`
>
> This will never have unicode (these are just string values of the dotted
> representation. The text discussion is for the values of Names since they
> can contain unicode). Everywhere else in our docs (e.g. name on cipher
> algorithms) we call this type str.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1645/files#r24108981>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084